### PR TITLE
10/deposit page widget

### PR DESCRIPTION
**Pull request by alfetopito**
Implemented static Deposit widget defined on *RF.3.2 Deposit Widget: Layout*. (https://github.com/gnosis/dex-react/issues/10)

Bear with me on the styling. Needles to mention, CSS is definitely not my strongest skill :|

Currently looks like this:

DepositWidget

Feel free to pick on each and every detail. Happy to adjust to however the team is used to do things.

### CODE REVIEW

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr20--dexreact.review.gnosisdev.com

File: src/components/DepositWidget/Table.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import React from 'react'
+import styled from 'styled-components'
+import { Row, RowProps } from './Row'
+
+const TableWrapper = styled.table`
+  display: table;
+  flex-direction: colum;
```
Comment (anxolin) [inline src/components/DepositWidget/Table.tsx]: It's already a table, why display is table? 
Also, flex-direction is not for flex displays?

File: src/components/layout/index.tsx [hunk 2]
```diff
@@ -12,6 +12,7 @@ const Wrapper = styled.div`
   main {
     flex: 1;
     padding: 4rem 3rem 10rem 3rem;
+    margin: auto;
```
Comment (anxolin) [inline src/components/layout/index.tsx]: what does this do in the main content?

File: src/components/DepositWidget/Table.tsx (hunk 1 above)
Comment (alfetopito) [reply src/components/DepositWidget/Table.tsx]: CSS copy/paste from other style. Will remove.

File: src/components/layout/index.tsx (hunk 2 above)
Comment (alfetopito) [reply src/components/layout/index.tsx]: Based on https://css-tricks.com/snippets/css/a-guide-to-flexbox/:

_Example section, first example_
> This relies on the fact a margin set to `auto` in a flex container absorb extra space. So setting a vertical margin of auto will make the item perfectly centered in both axes.

Trade page for instance is exactly the same:

Trade

Although Source code page is now centered...

SourceCode

Any suggestions on how to center the Widget component?

File: src/components/DepositWidget/Row.tsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+export interface RowProps {
+  tokenLogo: string
+  tokenName: string
+  exchangeWallet: number
+  pendingDeposits: number
+  pendingWithdraws: number
+}
+
+export class Row extends React.Component<RowProps> {
```
[the remark this answers is not in the document]
Comment (alfetopito) [reply src/components/DepositWidget/Row.tsx]: Ok. I'll then remove the `Button` component and keep the `Row`. I believe `Row` will be complex enough to deserve its own component.

File: src/components/DepositWidget/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+        break
+      }
+      case ButtonTypes.ENABLE: {
+        text = 'Enable ' + this.props.tokenName
+        break
+      }
+    }
+
+    return (
+      <WrapperButton {...this.props}>
```
[the remark this answers is not in the document]
Comment (alfetopito) [reply src/components/DepositWidget/Button.tsx]: Will remove.

File: src/components/DepositWidget/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  ENABLE = 'enable'
+}
+
+export class Button extends React.Component<ButtonProps> {
+  constructor(props: ButtonProps) {
+    super(props)
+  }
+
+  render() {
+    let text
```
[the remark this answers is not in the document]
Comment (alfetopito) [reply src/components/DepositWidget/Button.tsx]: Will extract.

File: src/components/DepositWidget/Button.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+  background-color: var(--color-${props => props.btnType}-btn-bg)
+`
+
+export enum ButtonTypes {
+  DEPOSIT = 'deposit',
+  WITHDRAW = 'withdraw',
+  ENABLE = 'enable'
+}
+
+export class Button extends React.Component<ButtonProps> {
```
[the remark this answers is not in the document]
Comment (alfetopito) [reply src/components/DepositWidget/Button.tsx]: Will do.

File: src/components/layout/index.tsx (hunk 2 above)
Comment (anxolin) [reply src/components/layout/index.tsx]: Aha, ok. I understand.

I think for the widget we can just use a negative top margin and centered inside `main`. 

I mean, I would do `main` ocupate all the with, and then the widget can just define it's size, be centered, and use a negative `margin-top`

**Comment by gnosis-info** [comment]
Travis automatic deployment:
 https://pr20--dexreact.review.gnosisdev.com

**Comment by alfetopito** [comment]
Comments addressed on latest revision.
Also, updated styling on buttons (although was not able to make them aligned). More work still required, committing them now to not be blocked.

NewButtons

**Pull request merged**